src/thunderbots/software/util/parameter/parameter.h

```diff
@@ -71,11 +72,50 @@ class Parameter
         if (Parameter<T>::getMutableRegistry().count(this->name_))
         {
             auto& param_in_registry = Parameter<T>::getMutableRegistry().at(this->name_);
-            return param_in_registry->value_;
+
+            param_in_registry.first->lock();
+            auto value = param_in_registry.second->value_;
+            param_in_registry.first->unlock();
+
+            return value;
+        }
+
+        // TODO fix this on ROS removal to throw an exception, ideally we do not
+        // want to return a value here as params that aren't in the registry are not
+        // threadsafe
+        else
+        {
+            return this->value_;
         }
         // if the parameter hasn't been registered yet, return default value
         return this->value_;
     }
+
+    /**
+     * Given the value, sets the value of this parameter
+     *
+     * @param new_value The new value to set
+     */
+    void setValue(const T new_value)
+    {
+        // get the value from the parameter in the registry
+        if (Parameter<T>::getMutableRegistry().count(this->name_))
+        {
+            auto& param_in_registry = Parameter<T>::getMutableRegistry().at(this->name_);
+            param_in_registry.first->lock();
+            param_in_registry->value_ = new_value;
+            param_in_registry.first->unlock();
```

Comment:
Should use `scoped_lock` here as well: https://en.cppreference.com/w/cpp/thread/scoped_lock

Reply:
fixed